pytorch_lightning/loggers/wandb.py

```diff
@@ -190,7 +190,14 @@ def experiment(self) -> Run:
         if self._experiment is None:
             if self._offline:
                 os.environ["WANDB_MODE"] = "dryrun"
-            self._experiment = wandb.init(**self._wandb_init) if wandb.run is None else wandb.run
+            if wandb.run is None:
+                self._experiment = wandb.init(**self._wandb_init)
+            else:
+                warning_cache.warn(
```

Comment:
Why do we need the warning cache if this will only runk on a single rank?

Reply:
warning cache is already used in a few places here in this logger. 
probably for jupyter notebook when they re-run the cell it would not show the warning multiple times. not sure.

Reply:
> probably for jupyter notebook when they re-run the cell it would not show the warning multiple times

wouldn't you want to still show the warning if re-running the cell?

> already used in a few places here in this logger

it was initially introduced in #5931. @borisdayma did you have a specific reason?

Reply:
I think there are no more reasons for it.
It was used when we were dropping values in the past due to `global_step` (not an issue anymore).
Then I kept it for the rest mainly t o have the right formatting of display (and also because it was already there).

I agree that you would want to show the warning again in notebooks.

Reply:
see #8764